bundles/catalogue/metadatasearch/instance.js

```diff
@@ -0,0 +1,403 @@
+import { MetadataStateHandler } from './MetadataStateHandler';
+import { Messaging } from 'oskari-ui/util';
+
+/**
+ * @class Oskari.mapframework.bundle.metadatasearch.MetadataSearchBundleInstance
+ *
+ * Main component and starting point for the "metadata catalogue" functionality.
+ * Provides metadata catalogue search functionality for the map.
+ *
+ * See Oskari.mapframework.bundle.metadatasearch.MetadataCatalogueBundle for bundle definition.
+ *
+ */
+export const METADATA_BUNDLE_LOCALIZATION_ID = 'catalogue.bundle.metadatasearch';
+const COVERAGE_FEATURE_STYLE = {
+    stroke: {
+        color: 'rgba(211, 187, 27, 0.8)',
+        width: 2
+    },
+    fill: {
+        color: 'rgba(255,222,0, 0.6)'
+    }
+};
+
+Oskari.clazz.define(
+    'Oskari.catalogue.bundle.metadatasearch.MetadataSearchBundleInstance',
+
+    /**
+     * @method create called automatically on construction
+     * @static
+     */
+    function () {
+        const sandboxName = (this.conf ? this.conf.sandbox : null) || 'sandbox';
+        this.sandbox = Oskari.getSandbox(sandboxName);
+        this.sandbox.register(this);
+        this.started = false;
```

Comment:
This is usually done on start() instead and I'm not sure if started variable is actually needed, but I recognize it from other bundles so it's probably copy-pasted here.

Reply:
Yeah, it's copypasted alright. And it is set to true at start method unless it's already set to true, in which case the start() - method does nothing. Wasn't sure of the side-effects this might have so I left it as it was.